go/carmen/block.go

```diff
@@ -137,6 +139,27 @@ func (c *archiveBlockContext) RunTransaction(run func(TransactionContext) error)
 	return runTransaction(c, run)
 }
 
+func (c *archiveBlockContext) GetProof(address Address, keys ...Key) (WitnessProof, error) {
+	c.lock.Lock()
+	if c.db == nil {
+		c.lock.Unlock()
+		return WitnessProof{}, fmt.Errorf("cannot get proof in invalid block context")
+	}
+	c.lock.Unlock()
```

Comment:
I think this lock should be held until the end since otherwise the database underlying the `archiveState` may get closed while running this function.

Reply:
I guess we should prevent closing of database when a witness proof creation is running, similarly as we do for the head block state, because locking the whole method may be too coarse?

Reply:
Actually it seems it is already in-place:
https://github.com/Fantom-foundation/Carmen/blob/main/go/carmen/database.go#L207
and
https://github.com/Fantom-foundation/Carmen/blob/main/go/carmen/database.go#L264

Reply:
Alright, this may allow multiple proof-queries in parallel.